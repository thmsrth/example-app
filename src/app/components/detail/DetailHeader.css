.DetailHeader {
  flex-grow: 1;
  box-shadow: var(--section-box-shadow);
  padding: var(--section-header-padding);
  margin: var(--section-header-margin);
  background-color: var(--section-background-color);
  border-top: var(--section-border-top);
  border-radius: var(--section-border-radius);
}

.DetailHeader__container {
  height: 100%;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: flex-start;
}

.DetailHeader__header {
  display: flex;
  align-items: center;
}

.DetailHeader__title {
  display: flex;
  align-items: baseline;
  margin: 0;
}

.DetailHeaderBack {
  margin-right: var(--space);
  color: var(--action-show-color);
}

.DetailHeaderBack__button__text,
.DetailHeaderBack__button__icon {
  color: var(--action-show-color);
}

.DetailHeader__actions {
  display: inline-flex;
  justify-content: flex-end;
  flex-grow: 1;
}

.DetailHeader__button {
  margin-left: var(--section-margin);
}
