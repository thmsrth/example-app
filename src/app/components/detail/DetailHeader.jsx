import React from 'react';

import PropTypes from 'prop-types';
import filtersValuesType from '../../../global/components/Filter/types/filtersValues';
import dataItemType from '../types/dataItem';

import { MAIN_PATH } from '../../../../config/constants';

import { historyPush } from '../../../global/utils/UrlHelpers';
import Button from '../../../global/components/Button';

import I18n from '../../utils/I18n';

import DetailHeaderActionsWrapper from './DetailHeaderActionsWrapper';

import './DetailHeader.css';

const DetailHeader = ({
  item, dataPageSize, dataPageNumber, dataSortParam, dataFilters, ...otherProps
}) => {
  if (item) {
    const queryParams = {
      size: dataPageSize,
      page: dataPageNumber,
      sort: dataSortParam,
      ...dataFilters,
    };
    const { referenceId } = item;

    return (
      <div className="DetailHeader">
        <div className="DetailHeader__container">
          <div className="DetailHeader__header">
            <div className="DetailHeaderBack">
              <Button
                className="DetailHeaderBack__button"
                iconName="arrow-circle-left"
                text={I18n.t('en.actions.back')}
                onClick={() => historyPush(MAIN_PATH, queryParams)}
              />
            </div>
            <h2 className="DetailHeader__title">
              {referenceId}
            </h2>
          </div>
          <div className="DetailHeader__actions">
            <DetailHeaderActionsWrapper
              item={item}
              {...otherProps}
            />
          </div>
        </div>
      </div>
    );
  }
  return null;
};

export default DetailHeader;

DetailHeader.propTypes = {
  item: dataItemType,
  dataPaginationSize: PropTypes.number,
  dataPaginationOffset: PropTypes.number,
  dataSortParam: PropTypes.string,
  dataFilters: filtersValuesType,
};

DetailHeader.defaultProps = {
  item: {},
  dataPaginationSize: 25,
  dataPaginationOffset: 0,
  dataSortParam: 'id,ASC',
  dataFilters: {},
};
