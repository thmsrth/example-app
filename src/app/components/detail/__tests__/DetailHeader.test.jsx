import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import DetailHeader from '../DetailHeader';

global.permissions = {};

const item = {};
const props = { item };

describe('DetailHeader', () => {
  it('should render the component correctly', () => {
    const tree = shallow(<DetailHeader {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
