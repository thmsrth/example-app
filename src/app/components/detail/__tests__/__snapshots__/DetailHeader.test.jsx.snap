// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailHeader should render the component correctly 1`] = `
<div
  className="DetailHeader"
>
  <div
    className="DetailHeader__container"
  >
    <div
      className="DetailHeader__header"
    >
      <div
        className="DetailHeaderBack"
      >
        <ButtonWrapper
          className="DetailHeaderBack__button"
          iconName="arrow-circle-left"
          onClick={[Function]}
          params={Object {}}
          text="Back"
        />
      </div>
      <h2
        className="DetailHeader__title"
      />
    </div>
    <div
      className="DetailHeader__actions"
    >
      <DetailHeaderActionsWrapper
        dataPaginationOffset={0}
        dataPaginationSize={25}
        deleteData={[Function]}
        item={Object {}}
      />
    </div>
  </div>
</div>
`;
